File: openapi-test-java-server-jaxrs-client-jdk/testapi/model/Email.java
package testapi.model;

import com.fasterxml.jackson.annotation.JsonValue;

/** Email address wrapper */
public record Email(@JsonValue String value) {
  public Email withValue(String value) {
    return new Email(value);
  };

  @Override
  public String toString() {
    return value;
  };
}

File: openapi-test-scala-server-spring-client-jdk/testapi/model/Email.scala
package testapi.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

@JsonCreator
/** Email address wrapper */
case class Email(@JsonValue value: String) extends AnyVal

object Email {

  /** Converts a String to this type for Spring @PathVariable binding */
  def valueOf(s: String): Email = new testapi.model.Email(s)
}
